rpcserver.go

```diff
@@ -6565,29 +6571,32 @@ func (r *rpcServer) BakeMacaroon(ctx context.Context,
 	}
 
 	// Validate and map permission struct used by gRPC to the one used by
-	// the bakery.
+	// the bakery. If the --allow-external-permissions flag is set, we
+	// will not validate, but map.
 	requestedPermissions := make([]bakery.Op, len(req.Permissions))
 	for idx, op := range req.Permissions {
-		if !stringInSlice(op.Entity, validEntities) {
-			return nil, fmt.Errorf("invalid permission entity. %s",
-				helpMsg)
-		}
-
-		// Either we have the special entity "uri" which specifies a
-		// full gRPC URI or we have one of the pre-defined actions.
-		if op.Entity == macaroons.PermissionEntityCustomURI {
-			allPermissions := r.interceptorChain.Permissions()
-			_, ok := allPermissions[op.Action]
-			if !ok {
-				return nil, fmt.Errorf("invalid permission " +
-					"action, must be an existing URI in " +
-					"the format /package.Service/" +
-					"MethodName")
+		if !req.AllowExternalPermissions {
```

Comment:
style nit: add a `continue` here to avoid an extra indentation.
For example:

```
for idx, op := range req.Permissions {
    if !req.AllowExternalPermissions {
        continue
    }
    ...
}
```